### PR TITLE
FEAT: Scout Talents for Sponsors!

### PR DESCRIPTION
**Pull request by JayeshVP24**
## Screenshots (if appropriate)

### CODE REVIEW

File: src/pages/dashboard/listings/[slug]/submissions.tsx [hunk 1]
```diff
@@ -21,25 +37,33 @@ interface Props {
...
+
 function BountySubmissions({ slug }: Props) {
   const router = useRouter();
   const { isOpen, onOpen, onClose } = useDisclosure();
   const { userInfo } = userStore();
   const [bounty, setBounty] = useState<Bounty | null>(null);
-  const [totalSubmissions, setTotalSubmissions] = useState(0);
-  const [totalWinners, setTotalWinners] = useState(0);
-  const [totalPaymentsMade, setTotalPaymentsMade] = useState(0);
+  const [totalSubmissions, setTotalSubmissions] = useState(-1);
```
Comment (JayeshVP24) [inline src/pages/dashboard/listings/[slug]/submissions.tsx:49]: i dont remember doing these 0 -> -1, is this a auto merge error?

Comment (a20hek) [reply src/pages/dashboard/listings/[slug]/submissions.tsx:49]: no..

File: src/layouts/Sponsor.tsx [hunk 2]
```diff
@@ -73,13 +79,49 @@ export function Sidebar({
...
     }
+  };
+
+  useEffect(() => {
+    const modalsToShow = async () => {
+      if (
+        userInfo?.currentSponsorId &&
+        (!userInfo?.firstName || !userInfo?.lastName || !userInfo?.username)
+      ) {
+        onScoutAnnounceModalClose();
```
Comment (a20hek) [inline src/layouts/Sponsor.tsx:110]: why close the modal? opening it only if all conditions are met is a better idea right?

File: src/pages/dashboard/listings/[slug]/submissions.tsx (hunk 1 above)
Comment (a20hek) [reply src/pages/dashboard/listings/[slug]/submissions.tsx:49]: change them back to 0

File: src/pages/dashboard/listings/[slug]/submissions.tsx [hunk 3]
```diff
@@ -21,25 +37,33 @@ interface Props {
...
   const [submissions, setSubmissions] = useState<SubmissionWithUser[]>([]);
+  const [scouts, setScouts] = useState<ScoutRowType[]>([]);
   const [selectedSubmission, setSelectedSubmission] =
     useState<SubmissionWithUser>();
   const [rewards, setRewards] = useState<string[]>([]);
   const [isBountyLoading, setIsBountyLoading] = useState(true);
-  const [skip, setSkip] = useState(0);
-  let length = 10;
+  const [skip, setSkip] = useState(-1);
+  let length = 9;
```
Comment (a20hek) [inline src/pages/dashboard/listings/[slug]/submissions.tsx:59]: here too

**Review by a20hek — CHANGES_REQUESTED** [review]

File: src/pages/dashboard/listings/[slug]/submissions.tsx [hunk 4]
```diff
@@ -129,115 +186,174 @@ function BountySubmissions({ slug }: Props) {
...
+                          (scout) => scout.userId === userId,
+                        );
+                        if (!scout) return;
+                        if (scout) {
+                          scout.invited = true;
+                        }
+                        const scoutIndex = scouts.findIndex(
+                          (scout) => scout.userId === userId,
+                        );
+                        if (scoutIndex > -1 && scoutIndex < scouts.length) {
```
Comment (a20hek) [inline src/pages/dashboard/listings/[slug]/submissions.tsx:344]: so in this file, all numbers got subtracted by 1 for some reason. hence asking you: is the `-1` here intentional?

Comment (JayeshVP24) [reply src/pages/dashboard/listings/[slug]/submissions.tsx:344]: all other numbers being subtracted one wasn't done by me, 
- but here findIndex returns -1 if element not found.

File: src/layouts/Sponsor.tsx (hunk 2 above)
Comment (JayeshVP24) [reply src/layouts/Sponsor.tsx:110]: yea same thing, must've missed to remove this line

Comment (JayeshVP24) [reply src/layouts/Sponsor.tsx]: u are right, pushed to remove this

**Pull request merged**